NullPointer exception while trying to use swagger-codegen-maven-plugin 2.2.2-SNAPSHOT to generate Spring server

<!--
Please follow the issue template below for bug reports and feature requests.
Also please indicate in the issue title which language/library is concerned. Eg:  [JAVA] Bug generating foo with bar 
-->

##### Description
I'm trying to generate Spring server-side sources from swagger.json and I'm consistently getting a NullPointerException after all files have been generated.

> [ERROR] Failed to execute goal io.swagger:swagger-codegen-maven-plugin:2.2.2-SNAPSHOT:generate (generate-server) on project adapter: Execution generate-server of goal io.swagger:swagger-codegen-maven-plugin:2.2.2-SNAPSHOT:generate failed. NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.swagger:swagger-codegen-maven-plugin:2.2.2-SNAPSHOT:generate (generate-server) on project adapter: Execution generate-server of goal io.swagger:swagger-codegen-maven-plugin:2.2.2-SNAPSHOT:generate failed.
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution generate-server of goal io.swagger:swagger-codegen-maven-plugin:2.2.2-SNAPSHOT:generate failed.
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 20 more
Caused by: java.lang.NullPointerException
	at io.swagger.codegen.plugin.CodeGenMojo.execute(CodeGenMojo.java:348)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	... 21 more

##### Swagger-codegen version

2.2.2-SNAPSHOT, 9f8a1e27ddedfe83db014058154f572d66b5bb59

##### Steps to reproduce

Use swagger-codegen-maven-plugin 2.2.2-SNAPSHOT to generate Spring server.

`{
  "allowUnicodeIdentifiers": "true",
  "basePackage": "<foo>",
  "apiPackage": "<bar>",
  "modelPackage": "<baz>",
  "invokerPackage": "<daz>",
  "groupId": "<fooz>",
  "artifactId": "<barz>",
  "artifactVersion": "0.0.1",
  "dateLibrary": "java8",
  "title": "Autogenerated Server",
  "interfaceOnly": false,
  "java8": true,
  "serializableModel": false,
  "async": false,
  "library": "spring-boot"
}
`

`
            <plugin>
                <groupId>io.swagger</groupId>
                <artifactId>swagger-codegen-maven-plugin</artifactId>
                <version>2.2.2-SNAPSHOT</version>
                <executions>
                    <execution>
                        <id>generate-server</id>
                        <phase>process-resources</phase>
                        <goals>
                            <goal>generate</goal>
                        </goals>
                        <configuration>
                            <inputSpec>${project.basedir}/out/${swagger.name}</inputSpec>
                            <configurationFile>${project.basedir}/src/resources/${config.name}</configurationFile>
                            <templateDirectory>${project.basedir}/src/resources/template/JavaSpring/</templateDirectory>
                            <language>spring</language>
                            <output>${generated-server.path}</output>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
`

--- a/modules/swagger-codegen-maven-plugin/src/main/java/io/swagger/codegen/plugin/CodeGenMojo.java
+++ b/modules/swagger-codegen-maven-plugin/src/main/java/io/swagger/codegen/plugin/CodeGenMojo.java
@@ -345,7 +345,7 @@ public class CodeGenMojo extends AbstractMojo {
         }
 
         if (addCompileSourceRoot) {
-            final Object sourceFolderObject = configOptions.get(CodegenConstants.SOURCE_FOLDER);
+            final Object sourceFolderObject = configOptions == null ? null : configOptions.get(CodegenConstants.SOURCE_FOLDER);
             final String sourceFolder =  sourceFolderObject == null ? "src/main/java" : sourceFolderObject.toString();
 
             String sourceJavaFolder = output.toString() + "/" + sourceFolder;

